.github/workflows/wheel.yml

```diff
@@ -2,88 +2,125 @@ name: Build binary wheel
-        python-version: [36, 37, 38, 39, 310]
+        include: ${{ fromJson(needs.build_wheels_matrix.outputs.include) }}
+
+    permissions:
+      contents: write # for uploading release assets
 
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v4
       with:
         submodules: true
 
     - name: Copy dll
-      if: (matrix.os == 'windows-latest')
+      if: matrix.os == 'windows-latest'
       uses: ./ci/actions/windll
 
-    - uses: actions/setup-python@v2
-      name: Install Python
-      with:
-        python-version: '3.7'
-
-    - name: Install cibuildwheel
-      run: |
-        python -m pip install cibuildwheel==2.8.1
-
-    - name: Install twine/wheel
-      run: |
-        python -m pip install twine wheel
-
     - name: chores
       if: matrix.os != 'windows-latest'
       run: |
-        mkdir dist
         mkdir packages/vaex-core/bin
         cp bin/install_pcre.sh packages/vaex-core/bin/
 
-    - name: Build vaex-core (windows)
-      if: (matrix.os == 'windows-latest')
-      env:
-        CIBW_BUILD: cp${{ matrix.python-version }}*64
-        CIBW_ENVIRONMENT_LINUX: "CFLAGS='-Wl,-strip-all' CXXFLAGS='-Wl,-strip-all'"
-      run: |
-        python -m cibuildwheel --output-dir dist packages/vaex-core
+    - name: Set up QEMU
+      if: runner.os == 'Linux'
+      uses: docker/setup-qemu-action@v3
 
-    - name: Build vaex-core (linux, osx)
-      if: (matrix.os != 'windows-latest')
+    - uses: pypa/cibuildwheel@v2.17.0 # sync version with pip install cibuildwheel above
+      with:
+        only: ${{ matrix.only }}
+        package-dir: packages/vaex-core/
+        output-dir: packages/vaex-core/dist/
       env:
-        CIBW_BEFORE_BUILD: bin/install_pcre.sh
-        CIBW_BUILD: "cp${{ matrix.python-version }}*64 cp${{ matrix.python-version }}-macosx-*"
-        CIBW_ENVIRONMENT_LINUX: "CFLAGS='-Wl,-strip-all' CXXFLAGS='-Wl,-strip-all'"
+        CIBW_BEFORE_BUILD: ${{ startswith(matrix.os, 'ubuntu') && 'bash bin/install_pcre.sh' || startswith(matrix.os, 'macos') && 'sudo -E bash bin/install_pcre.sh' || '' }}
         CIBW_BUILD_VERBOSITY: 2
-        # needed for m1 builds
+        CIBW_MANYLINUX_X86_64_IMAGE: manylinux_2_28
+        CIBW_MANYLINUX_AARCH64_IMAGE: manylinux_2_28
+        CIBW_ENVIRONMENT_LINUX: "CFLAGS='-Wl,-strip-all' CXXFLAGS='-Wl,-strip-all'"
         CIBW_ENVIRONMENT_MACOS: "CFLAGS='-I/usr/local/include -L/usr/local/lib' CXXFLAGS='-I/usr/local/include -L/usr/local/lib' LDFLAGS='-L/usr/local/lib'"
-        CIBW_ARCHS_MACOS: "x86_64 arm64"
-      run: |
-        python -m cibuildwheel --output-dir dist packages/vaex-core
 
-    - name: Make source distribution
-      shell: bash
-      run: |
-          cd packages/vaex-core && python setup.py sdist
+    - name: Upload release assets
+      if: github.event_name == 'release'
+      uses: softprops/action-gh-release@v0.1.15
+      with:
+        files: packages/vaex-core/dist/*
```

Comment:
We are not using the release feature of github, but release on tags, looking at the docs, it seems we should have a similar if as 'Publish a Python distribution to PyPI' right?

Reply:
this step will upload wheels to a github release, eg https://github.com/ddelange/python-magic/releases/tag/0.4.28.post7

it is a no-op if there is no release